superset/utils/core.py

```diff
@@ -963,8 +974,10 @@ def send_email_smtp(  # pylint: disable=invalid-name,too-many-arguments,too-many
         image.add_header("Content-ID", "<%s>" % msgid)
         image.add_header("Content-Disposition", "inline")
         msg.attach(image)
-
-    send_mime_email(smtp_mail_from, recipients, msg, config, dryrun=dryrun)
+    msg_mutator = config["EMAIL_HEADER_MUTATOR"]
+    # the base notification returns the message without any editing.
+    new_msg = msg_mutator(msg, **header_data or {})
```

Comment:
would it be easier to have header_data default to an empty dictionary on line 916? I think this might still fail if `header_data` were None.

Reply:
Yeah! This is something that I learned, you actually are not allowed to make a default into an empty dictionary or list.